sycl/doc/syclcompat/README.md

```diff
@@ -0,0 +1,1295 @@
+pitched_data malloc(sycl::range<3> size, sycl::queue q = get_default_queue());
+
+// Blocking memcpy
+void memcpy(void *to_ptr, const void *from_ptr, size_t size,
+            sycl::queue q = get_default_queue());
+void memcpy(T *to_ptr, const T *from_ptr, size_t count,
+            sycl::queue q = get_default_queue());
+void memcpy(void *to_ptr, size_t to_pitch, const void *from_ptr,
+            size_t from_pitch, size_t x, size_t y,
+            sycl::queue q = get_default_queue()); // 2D matrix
+void memcpy(pitched_data to, sycl::id<3> to_pos,
+            pitched_data from, sycl::id<3> from_pos,
+            sycl::range<3> size,
+            sycl::queue q = get_default_queue()); // 3D matrix
+
+// Non-blocking memcpy
+sycl::event memcpy_async(void *to_ptr, const void *from_ptr, size_t size,
+                         sycl::queue q = get_default_queue());
+template <typename T>
+sycl::event memcpy_async(T *to_ptr, T void *from_ptr, size_t count,
+                         sycl::queue q = get_default_queue());
+sycl::event memcpy_async(void *to_ptr, size_t to_pitch,
+                         const void *from_ptr, size_t from_pitch,
+                         size_t x, size_t y,
+                         sycl::queue q = get_default_queue()); // 2D matrix
+sycl::event memcpy_async(pitched_data to, sycl::id<3> to_pos,
+                         pitched_data from, sycl::id<3> from_pos,
+                         sycl::range<3> size,
+                         sycl::queue q = get_default_queue()); // 3D matrix
+
+// Fill
+template <class T>
+void fill(void *dev_ptr, const T &pattern, size_t count,
+          sycl::queue q = get_default_queue());
+template <typename T>
+sycl::event fill_async(void *dev_ptr, const T &pattern,
+                       size_t count, sycl::queue q = get_default_queue());
+
+// Memset
+void memset(void *dev_ptr, int value, size_t size,
+                   sycl::queue q = get_default_queue());
+void memset(void *ptr, size_t pitch, int val, size_t x, size_t y,
+            sycl::queue q = get_default_queue()); // 2D matrix
+void memset(pitched_data pitch, int val, sycl::range<3> size,
+                          sycl::queue q = get_default_queue()); // 3D matrix
+sycl::event memset_async(void *dev_ptr, int value, size_t size,
+                         sycl::queue q = get_default_queue());
+sycl::event memset_async(void *ptr, size_t pitch, int val,
+                         size_t x, size_t y,
+                         sycl::queue q = get_default_queue()); // 2D matrix
+sycl::event memset_async(pitched_data pitch, int val,
+                         sycl::range<3> size,
+                         sycl::queue q = get_default_queue()); // 3D matrix
+
+void free(void *ptr, sycl::queue q = get_default_queue());
+sycl::event free_async(const std::vector<void *> &pointers,
+                       const std::vector<sycl::event> &events,
+                       sycl::queue q = get_default_queue());
+
+/// Queries pointer allocation type
```

Comment:
```suggestion
// Queries pointer allocation type
```
Is the Doxygen style comment intentional here?

Reply:
Good catch. We'll remove these elsewhere too.